meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -115,6 +117,61 @@ public void create(String key, Object data, MetaClientInterface.EntryMode mode)
     }
   }
 
+  @Override
+  public void createFullPath(String key, Object data, MetaClientInterface.EntryMode mode) {
+
+    boolean retry;
+    MetaClientInterface.EntryMode parentMode = (EntryMode.EPHEMERAL.equals(mode) ?
+        EntryMode.PERSISTENT : mode);
+    do {
+      retry = false;
+      try {
+        create(key, data, mode);
+      } catch (MetaClientNoNodeException e) {
+        retry = true;
+        String parentPath = getZkParentPath(key);
+        try {
+          createFullPath(parentPath, null, parentMode);
+        } catch (MetaClientNodeExistsException e1) {
+          return;
+        }
+      } catch (MetaClientNodeExistsException e) {
+        LOG.warn("Node already exists. path: " + key);
+        throw e;
+      } catch (MetaClientException e) {
+        LOG.error("Exception while creating path: " + key, e);
+        throw e;
+      }
+    } while (retry);
+  }
+
+  @Override
+  public void createFullPathWithTTL(String key, T data, long ttl) {
```

Comment:
These 2 logics seems a bit repetitive. We could have a helper function with options.

Reply:
Created a helper, recursiveCreateHelper. Let me know if you have any suggestions, especially with naming haha